cmake/thirdparty/FindMFEM.cmake

```diff
@@ -125,8 +125,18 @@ endif()
 message(STATUS "MFEM Includes: ${MFEM_INCLUDE_DIRS}")
 message(STATUS "MFEM Libraries: ${MFEM_LIBRARIES}")
 
-blt_register_library(
-    NAME          mfem
-    INCLUDES      ${MFEM_INCLUDE_DIRS}
-    LIBRARIES     ${MFEM_LIBRARIES}
-    TREAT_INCLUDES_AS_SYSTEM ON)
+if(ENABLE_CUDA)
+    message(STATUS "MFEM: Adding required CUDA includes/libraries")
+    blt_register_library(
+        NAME          mfem
+        INCLUDES      ${MFEM_INCLUDE_DIRS} ${CUDA_INCLUDE_DIRS}
+        # FIXME: How does this work with first-class CMake CUDA support
+        LIBRARIES     ${MFEM_LIBRARIES} ${CMAKE_CUDA_LINK_FLAGS} ${CUDA_LIBRARIES}
```

Comment:
Could you not use the BLT CUDA registered libraries? `cuda` or `cuda_runtime`

Reply:
I'm not sure that BLT-registered libraries can be used when registering a BLT library - I could be wrong though.

Either way, the runtime libraries alone are not sufficient because MFEM also requires `cusparse` but doesn't reference it in the `MFEM_EXT_LIBS` variable that we use to get the paths to the rest of its libraries.

Reply:
You can, just list it on the LIBRARIES line.  You should also be able to just add `cusparse`.

Reply:
Just to correct my previous statement... blt registered libraries should go on the DEPENDS_ON line.  /hangHeadInShame

Reply:
This will be handled in a cleaner way once https://github.com/LLNL/blt/pull/421 is merged